src/Groups/cosets.jl

```diff
@@ -874,9 +876,88 @@ function Base.in(g::GAPGroupElem, C::GroupCoset)
 end
 
 function Base.in(g::GAPGroupElem, C::GroupDoubleCoset)
-  return GapObj(g) in GapObj(C)
-#TODO: avoid delegation to GAP?
-# (GAP uses `RepresentativesContainedRightCosets`, `CanonicalRightCosetElement`)
+  if !isassigned(C.right_coset_reps)
+    C.right_coset_reps[] = _right_coset_reps(C)
+  end
+  canon = GAP.Globals.CanonicalRightCosetElement(GapObj(left_acting_group(C)), GapObj(g))
+  return haskey(C.right_coset_reps[], group_element(group(C), canon))
+end
+
+"""
+   _decompose(C::GroupDoubleCoset, x::GAPGroupElem)
+
+Return `flag, u, v` such that `flag` is `true` if `x` is an element
+of `C`, and `false` otherwise.
+
+If `flag = true` then `x = u*g*v` holds where `g` is `representative(C)`,
+`u` is an element of `left_acting_group(C)`,
+and `v` is an element of `right_acting_group(C)`.
+
+# Examples
+```jldoctest
+julia> G = symmetric_group(5);
+
+julia> H = sylow_subgroup(G, 2)[1]; K = sylow_subgroup(G, 3)[1];
+
+julia> x = gen(G, 1); C = double_coset(H, x, K);
+
+julia> d =Oscar._decompose(C, x^3)
+(true, (1,3)(2,4), (1,3,2))
+
+julia> d[2] * x * d[3] == x^3
+true
+
+julia> Oscar._decompose(C, x^2)
+(false, (), ())
+"""
+function _decompose(C::GroupDoubleCoset, x::GAPGroupElem)
+  G = group(C)
+  U = left_acting_group(C)
+  if !isassigned(C.right_coset_reps)
+    C.right_coset_reps[] = _right_coset_reps(C)
+  end
```

Comment:
I am not sure if Julia can correctly deduce that it has to fetch the value in `C.right_coset_reps[]` only once afterwards and then not again(in lines 922, 923).

All in all, I'd suggest to use a pattern like this:
```suggestion
  rcr = _right_coset_reps(C)
```
and then rename the current `_right_coset_reps` to, say, `compute_right_coset_reps`; and the new `_right_coset_reps` basically does 
```
function _right_coset_reps(C)
  if !isassigned(C.right_coset_reps)
    C.right_coset_reps[] = _right_coset_reps(C)
  end
  return C.right_coset_reps[]
end
```